mava/components/building/distributor.py

```diff
@@ -75,6 +78,14 @@ def on_building_program_nodes(self, builder: SystemBuilder) -> None:
         # Save number of the executors
         builder.store.num_executors = self.config.num_executors
 
+        # Save list of metrics attached with their best performance
+        builder.store.checkpointing_metric: Dict[str, Any] = {}  # type: ignore
```

Comment:
Should this not be in a checkpointing component?

Reply:
The issue here is that we want to share the metrics names between the executor in the env loop, the checkpointer and the parameter server so the best place to have it for all the three is the distributor

Reply:
I don't completely understand. Maybe we can have a chat :sweat_smile:

Reply:
Done!